docs/source/autoinspection.rst

```diff
@@ -0,0 +1,34 @@
+.. _autoinspection:
+
+================================================================================
+Autoinspection
+================================================================================
+
+It can be very convenient to have great expectations automatically review a \
+dataset and suggest expectations that may be appropriate. Currently, there \
+a very basic, but easily extensible, autoinspection capability available.
+
+Dataset objects have an `autoinspect` method which allows you to provide a \
+function that will evaluate a dataset object and add expectations to it. \
+By default `autoinspect` will call the autoinspect function \
+:func:`columns_exist <great_expectations.dataset.autoinspect.columns_exist>` \
+which will add an `expect_column_to_exist` expectation for each column \
+currently present on the dataset.
+
+To implement additional autoinspection functions, you simply take a single \
+parameter, a Dataset, and evaluate and add expectations to that object.
+
+
+.. code-block:: python
+
+    >> import great_expectations as ge
+    >> df = ge.dataset.PandasDataset({"col": [1, 2, 3, 4, 5]})
+    >> df.autoinspect(ge.dataset.autoinspect.columns_exist)
```

Comment:
FWIW, I think there's another mode that'll be useful for autoinspect: 
`ge.autoinspect(my_dataframe, ge.dataset.autoinspect.columns_exist)` -> Returns expectation_config in JSON format without altering `my_dataframe`

Worth implementing from the start?